spec.html

```diff
@@ -12305,15 +12305,15 @@ <h1>
     </h1>
     <dl class="header">
       <dt>description</dt>
-      <dd>It returns *true* if and only if _v_ is considered to have an identity suitable for a weak reference. This determines whether _v_ can be a key of a WeakMap, or entry in a WeakSet, or target of a WeakRef or FinalizationRegistry.</dd>
+      <dd>It returns *true* if and only if _v_ is suitable for use as a weak reference. Only values that are suitable for use as a weak reference may be a key of a WeakMap, an element of a WeakSet, the target of a WeakRef, or one of the targets of a FinalizationRegistry.</dd>
     </dl>
     <emu-alg>
       1. If _v_ is an Object, return *true*.
       1. If _v_ is a Symbol and KeyForSymbol(_v_) is *undefined*, return *true*.
       1. Return *false*.
     </emu-alg>
     <emu-note>
-      <p>Symbols in the GlobalSymbolRegistry (<emu-xref href="#sec-symbol.for"></emu-xref>) are not considered suitable for use as a WeakMap key because they can be described by the String used to register them, and therefore do not have <emu-xref href="#sec-identity">identity</emu-xref>. Well-known symbols (<emu-xref href="#sec-well-known-symbols"></emu-xref>) are likely to never be collected, but are nonetheless treated as suitable for weak reference because they are limited in number and therefore manageable by a variety of implementation approaches. However, any values associated to a well-known symbol in a live WeakMap is unlikely to be collected and could “leak” memory resources in many implementations.</p>
+      <p>Values without identity are not suitable for use as a weak reference because they may be manifest at any point without a prior reference. Symbols in the GlobalSymbolRegistry (<emu-xref href="#sec-symbol.for"></emu-xref>) are not suitable for use as a weak reference because they may be retrieved again at any later point using the String with which they were registered, making them ineligible for collection and therefore eternal. Well-known symbols (<emu-xref href="#sec-well-known-symbols"></emu-xref>) are likely to never be collected, but are nonetheless suitable for weak reference because they are limited in number and therefore manageable by a variety of implementation approaches. However, any value associated to a well-known symbol in a live WeakMap is unlikely to be collected and could "leak" memory resources in some implementations.</p>
```

Comment:
Michael, I understand that you prefer all Symbols to have identity but I don't think "may be retrieved again at any later point using the String" is helpful as a separate reason from "may be manifest at any point without prior reference". "May be manifest at any point without prior reference" already fully explains `Symbol.for`. The specification mechanism of manifestation is the String-keyed global registry, but the reason that neither Symbol.for symbols and values without identity are suitable for weak references is _exactly the same_.

I feel that explaining that singular reason as two separate reasons muddles the explanation.

I don't see clarity being lost if we say that Symbol.for symbols don't have identity but other Symbols do. It fits very well with the existing definition of identity as "manifestable without prior reference". It means Symbols can't be thought of neatly as one category of values wrt identity, but that is a fact of the language today because we designed a crappy feature.

Here's a thought experiment. Suppose the spec specified all String values to be interned with a cross-Realm registry, like is the case of Symbol.for symbols. There is no observable difference on the language, and is a specification detail. Would you then say Strings have identity? I wouldn't.

Reply:
> I feel that explaining that singular reason as two separate reasons muddles the explanation.

I'm fine with not elaborating on values without identity here. I added it for completeness and because I thought it might help with clarity. If you think it hurts clarity, I don't mind removing it.

> I don't see clarity being lost if we say that Symbol.for symbols don't have identity but other Symbols do.

The thing is, there's no such thing as Symbol.for symbols. There's just symbols. Sometimes they are in the GlobalSymbolRegistry, sometimes not. No symbols are in the GlobalSymbolRegistry at their inception. And there's no invariant described in the spec at the moment that, once added, they are never removed. This intersects with my desire for whether something has identity to be an unchanging property of that thing.

> but that is a fact of the language today because we designed a crappy feature

It's not. The formalisation I've proposed here does not rely on separation of these concepts or some exotic definition of identity.

> Would you then say Strings have identity?

As long as the interning was used for spec uses of strings as well, it wouldn't matter whether strings had identity or not in this scenario. It's only important for strings to not have identity in the spec currently because we want `"a" is "a"` to be `true`. I don't know what this was supposed to illustrate.